### PR TITLE
Random architecture changed 

### PR DESCRIPTION
**Pull request by Kubasinska**
All random choices of the model are made according to the rng object (defult_rng) contained in the extra module. If the seed is an integer or an array, the result becomes deterministic, otherwise a stochastic instance of the system is taken and the model becomes non-deterministic.

### CODE REVIEW

File: mesa/Employer.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def create_job(self,  level:int, worker: Person):
     def employees(self):
         return [x.worker for x in my_jobs]
 
-vjob_level:int, my_employer: Employer, my_worker: Person, m: mesaPROTON_OC.MesaPROTON_OC
```
Comment (mariopaolucci) [inline mesa/Employer.py:25]: Why is this uncommented?

File: mesa/Person.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Person(Agent):
         'professional',
         'school']      
     
-    def __init__(self, m: mesaPROTON_OC):
+    def __init__(self, m:mesaPROTON_OC):
```
Comment (mariopaolucci) [inline mesa/Person.py:24]: thanks. I still don't really understand how this way of declaring types is strict. I use it unsystematically - we have to decide. #172.

File: mesa/extra.py [hunk 3]
```diff
@@ -8,10 +8,11 @@
 import Person
 import numpy as np
 import random
-
+from numpy.random import default_rng
+#The random generator used in all modules BitGenerator(PCG64)
+rng = default_rng(42) #If None, unpredictable entropy will be pulled from the OS.
```
Comment (mariopaolucci) [inline mesa/extra.py:13]: The default must be none.

File: mesa/mesaPROTON_OC.py [hunk 4]
```diff
@@ -1,26 +1,23 @@
+from extra import *
 from mesa import Agent, Model
 from mesa.time import RandomActivation
 from mesa.datacollection import DataCollector
-import Person
 import pandas as pd
 import numpy as np
 import networkx as nx
-from numpy.random import default_rng
 # correct way of using numpy.random: https://numpy.org/devdocs/reference/random/index.html
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:8]: move this comment in extra?

File: mesa/testProton.py [hunk 5]
```diff
@@ -5,7 +5,6 @@
 
 @author: paolucci
```
Comment (mariopaolucci) [inline mesa/testProton.py:6]: qui c'e' il generatore vecchio?

File: mesa/Person.py (hunk 2 above)
Comment (Kubasinska) [reply mesa/Person.py:24]: No doubt they make it easier to read the code. I think it is useful to use them when the choice is forced, as in this case.

File: mesa/extra.py (hunk 3 above)
Comment (Kubasinska) [reply mesa/extra.py:13]: Correct, I put the comment for completeness.

File: mesa/testProton.py (hunk 5 above)
Comment (Kubasinska) [reply mesa/testProton.py:6]: Si, sto lavorando a questa parte.

Comment (mariopaolucci) [reply mesa/testProton.py:6]: Ok, allora per pulizia meglio finire prima del pull (o, se troppi cambiamenti, dichiarare nel pull: "random architecture changed part 1"

Comment (Kubasinska) [reply mesa/testProton.py:6]: D'accordo!

**Pull request merged**